### PR TITLE
Add room calls: invite, delete, join, leave, set*

### PR DESCRIPTION
**Pull request by reist**

### CODE REVIEW

**Comment by reist** [comment]
Silly Rubocop, making me ping-pong ignores...

File: lib/rocket_chat/messages/room.rb [hunk 1]
```diff
@@ -95,6 +203,11 @@ def room_option_hash(options)
         options.each { |key, value| new_hash[Util.camelize(key)] = value }
         new_hash
       end
+
+      def validate_attribute(attribute)
+        raise ArgumentError, "Unsettable attribute: #{attribute}" unless \
```
Comment (abrom) [inline lib/rocket_chat/messages/room.rb]: I'd suggest using `"Unsettable attribute: #{attribute || 'nil'}"` instead as calling `set_attr` with no args would result in `raise ArgumentError, 'Unsettable attribute: '`

File: lib/rocket_chat/messages/room.rb [hunk 2]
```diff
@@ -73,6 +162,25 @@ def rename(room_id, new_name)
...
+      # @return [Boolean]
+      # @raise [ArgumentError, HTTPError, StatusError]
+      #
+      def set_attr(room_id: nil, name: nil, **setting)
+        attribute, value = setting.first
+        validate_attribute(attribute)
+        session.request_json(
+          self.class.api_path(Util.camelize("set_#{attribute}")),
+          method: :post,
+          body: room_params(room_id, name)
```
Comment (abrom) [inline lib/rocket_chat/messages/room.rb]: Not related to this change directly, but would also make sense to raise an exception in `room_params` if neither id or name are set

**Comment by reist** [comment]
Awesome, thanks!

**Pull request merged**